### PR TITLE
IMAGEKIT-56 Typescript support

### PR DESCRIPTION
**Pull request by ahnv**
#26

### CODE REVIEW

File: src/interfaces/UploadResponse.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+   */
+  url: string;
+  /**
+   * In case of an image, a small thumbnail URL.
+   */
+  thumbnailUrl: string;
+  /**
+   * Height of the uploaded image file. Only applicable when file type is image.
+   */
+  height: number;
```
Comment (violetviolinist) [inline src/interfaces/UploadResponse.ts:36]: Are we sure the API returns Number JS values? if they're string, i.e. "123", then will it be coerced to TS's number?

File: test/initialization.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 const chai = require("chai");
 const expect = chai.expect;
 const initializationParams = require("./data").initializationParams
-import ImageKit from "../src/index.js";
+import ImageKit from "../out-tsc/src/index";
 
 
 describe("Initialization checks", function () {
+    console.log(ImageKit)
```
Comment (violetviolinist) [inline test/initialization.js]: redundant

Comment (ahnv) [reply test/initialization.js]: This is from an outdated commit

File: src/interfaces/UploadResponse.ts (hunk 1 above)
Comment (ahnv) [reply src/interfaces/UploadResponse.ts:36]: Yes, they are returned as number as per the documentation
https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload

File: src/interfaces/UploadOptions.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   *      Pass the full URL, for example - https://www.example.com/rest-of-the-image-path.jpg.
+   */
+  file: string | Buffer | File;
+  /**
+   * The name with which the file has to be uploaded.
+   * The file name can contain:
+   * - Alphanumeric Characters: a-z , A-Z , 0-9
+   * - Special Characters: . _ and -
+   * Any other character including space will be replaced by _
+   */
```
Comment (imagekitio) [inline src/interfaces/UploadOptions.ts:22]: This is outdated. Refer to docs for updated version

File: src/interfaces/UploadOptions.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   * - Comma-separated value of tags in format tag1,tag2,tag3. For example - t-shirt,round-neck,men
+   * - The maximum length of all characters should not exceed 500.
+   * - % is not allowed.
+   * - If this field is not specified and the file is overwritten then the tags will be removed.
+   */
+  tags?: string;
+  /**
+   * The folder path (e.g. /images/folder/) in which the image has to be uploaded. If the folder(s) didn't exist before, a new folder(s) is created.
+   * The folder name can contain:
+   * - Alphanumeric Characters: a-z , A-Z , 0-9
```
Comment (imagekitio) [inline src/interfaces/UploadOptions.ts:43]: This is outdated. Refer to docs for updated version

File: src/interfaces/UploadOptions.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+export interface UploadOptions {
+  /**
+   * This field accepts three kinds of values:
+   * - binary - You can send the content of the file as binary. This is used when a file is being uploaded from the browser.
+   * - base64 - Base64 encoded string of file content.
+   * - url - URL of the file from where to download the content before uploading.
+   *      Downloading file from URL might take longer, so it is recommended that you pass the binary or base64 content of the file.
+   *      Pass the full URL, for example - https://www.example.com/rest-of-the-image-path.jpg.
+   */
+  file: string | Buffer | File;
```
Comment (imagekitio) [inline src/interfaces/UploadOptions.ts:15]: What is `File`?

File: src/interfaces/UploadResponse.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+   */
+  isPrivateFile: boolean;
+  /**
+   * Value of custom coordinates associated with the image in format x,y,width,height.
+   */
+  customCoordinates: string | null;
+  /**
+   * The metadata of the upload file. Use responseFields property in request to get the metadata returned in response of upload API.
+   */
+  metadata?: string;
```
Comment (imagekitio) [inline src/interfaces/UploadResponse.ts:68]: `metadata` is not a string, it is an object.

**Review by imagekitio — CHANGES_REQUESTED** [review]

**Review by imagekitio — CHANGES_REQUESTED** [review]
This branch has conflicts that must be resolved,

File: src/interfaces/UploadOptions.ts (hunk 5 above)
Comment (ahnv) [reply src/interfaces/UploadOptions.ts:15]: `File` is a Browser Only API to upload files https://developer.mozilla.org/en-US/docs/Web/API/File

It is essentially a subclass of `Blob`

**Review by vaibhavbansal23 — APPROVED** [review]
fixed

**Pull request merged**